play-services-core/src/main/kotlin/org/microg/gms/credential/CredentialManagerService.kt

```diff
@@ -53,13 +61,27 @@ class CredentialManagerService : BaseService(TAG, GmsService.CREDENTIAL_MANAGER)
 
 private class CredentialManagerServiceImpl(private val context: Context, override val lifecycle: Lifecycle) : ICredentialManagerService.Stub(), LifecycleOwner {
 
-    override fun getCredentialManagerIntent(callback: IPendingIntentCallback?, params: CredentialManagerInvocationParams?) {
-        Log.d(TAG, "Not yet implemented: getCredentialManagerIntent $params")
+    override fun getCredentialManagerIntent(callback: IPendingIntentCallback?, params: CredentialManagerInvocationParams) {
+        Log.d(TAG, "Method getCredentialManagerIntent $params called")
         lifecycleScope.launchWhenStarted {
-            try {
+            runCatching {
+                val intent = Intent().apply {
+                    setClassName(Constants.GMS_PACKAGE_NAME, PASSWORD_MANAGER_CLASS_NAME)
+                    putExtra(EXTRA_KEY_ACCOUNT_NAME, params.account.name)
+                    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                    if (params.caller != null) {
+                        putExtra(EXTRA_KEY_UTM_SOURCE, params.caller.source)
+                        putExtra(EXTRA_KEY_UTM_MEDIUM, params.caller.medium)
+                        putExtra(EXTRA_KEY_UTM_CAMPAIGN, params.caller.campaign)
+                        putExtra(EXTRA_KEY_UTM_CONTENT, params.caller.content)
+                    }
+                }
+                val flags = PendingIntent.FLAG_ONE_SHOT or PendingIntent.FLAG_UPDATE_CURRENT
+                val pendingIntent = PendingIntent.getActivity(context, 0, intent, flags)
```

Comment:
If you use `FLAG_UPDATE_CURRENT` with a static `requestCode` (0), any other app calling this method will override the pending intent. e.g. if app A invokes this with account A and stores the returned pending intent for a brief while, than app B invokes this with account B, then app A invokes its stored pending intent, it will open account B.